### ISSUE TITLE
acceptance: allocator tests don't detect node crash during rebalance

### ISSUE THREAD

**Issue by cuongdo** [cockroachdb/cockroach#7691]
Example:
http://teamcity.cockroachdb.com/viewLog.html?buildId=913&buildTypeId=Cockroach_AllocatorTest1to310g&tab=buildLog

Relevant log entries:
```
[21:07:27][Step 1/1] I160707 01:07:27.703127 acceptance/allocator_test.go:292  last range event: add for range 286/store 1 (3.554787s ago)
[21:07:37][Step 1/1] E160707 01:07:37.706382 acceptance/allocator_test.go:283  error checking rebalancer: dial tcp 104.196.146.253:26257: getsockopt: connection refused
[21:07:47][Step 1/1] E160707 01:07:47.708143 acceptance/allocator_test.go:283  error checking rebalancer: dial tcp 104.196.146.253:26257: getsockopt: connection refused
```

Right now, the code that checks the rebalancer status naively continues after all errors. Instead, we need to continue after transient errors (e.g. context deadline exceeded) but fail on permanent errors (e.g. connection refused).

**Comment by tschottdorf** [comment]
Periodically calling Assert should also help, no? And requiring that there
is a non-error every minute.

On Thu, Jul 7, 2016, 23:47 Cuong Do <notifications@github.com> wrote:

> Example:
>
> http://teamcity.cockroachdb.com/viewLog.html?buildId=913&buildTypeId=Cockroach_AllocatorTest1to310g&tab=buildLog
>
> Relevant log entries:
>
> [21:07:27][Step 1/1] I160707 01:07:27.703127 acceptance/allocator_test.go:292  last range event: add for range 286/store 1 (3.554787s ago)
> [21:07:37][Step 1/1] E160707 01:07:37.706382 acceptance/allocator_test.go:283  error checking rebalancer: dial tcp 104.196.146.253:26257: getsockopt: connection refused
> [21:07:47][Step 1/1] E160707 01:07:47.708143 acceptance/allocator_test.go:283  error checking rebalancer: dial tcp 104.196.146.253:26257: getsockopt: connection refused
>
> We need to continue after transient errors (e.g. context deadline
> exceeded) but fail on permanent errors (e.g. connection refused).
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/issues/7691>, or mute the thread
> <https://github.com/notifications/unsubscribe/AE135CRcurWT1yMcRqNkdK7xArKOwYT8ks5qTchRgaJpZM4JHsxr>
> .
>
-- 

-- Tobias

**Comment by cuongdo** [comment]
fixed by #7772
